### DISCUSSION TITLE
Can't set a custom UDP port in firewall. Not sure if this is a bug or not.

### DISCUSSION THREAD

**Discussion by rkdawenterprises** [cockpit-project/cockpit#18326] [General]
Tried to open a UDP port 22222 but it won't let me. Gives me an error, "Port number and type do not match". I can try to add an ID "custom--yadayada" and description but that does not help. Am I doing it wrong? I ended up having to do it CLI with "firewall-cmd" into FirewallServer zone. If this is a bug, I will happily add it to the issues list and try and fix it.
![Screenshot from 2023-02-10 17-08-10](https://user-images.githubusercontent.com/108582693/218221617-e55d5636-099a-497d-8dc7-89a5f0a217c5.png)

**Comment by marusak** [comment]
What we do is that we load content of `/etc/services` and then we check if specified ports are known values.

```
$ grep 22222 /etc/services 
easyengine      22222/tcp               # EasyEngine is CLI tool to manage WordPress Sites on Nginx server
```

In this case it is known value for `easyengine` service that is only defined as tcp.

Question is whether we should prohibit users from using this port as udp.

**Reply by garrett** [reply]
It's not intended to use that `/etc/services` as an allow list or to use it even strictly. It's supposed to only be a guideline for guessing that services might be.

The firewall page is supposed to allow any port and protocol. If it doesn't allow it, like it looks like here, then this is most definitely a bug.

**Comment by almyp** [comment]
This got me as well. I need to enable 9093 with UDP and can’t.

**Comment by marusak** [comment]
This is actually a bug so I opened https://github.com/cockpit-project/cockpit/issues/18412
